pycbc/strain/strain.py

```diff
@@ -1374,7 +1374,7 @@ def __init__(self, frame_src, channel_name, start_time,
         self.psd = None
         self.psds = {}
 
-        strain_len = int(sample_rate * self.raw_buffer.delta_t * len(self.raw_buffer))
+        strain_len = int(max_buffer * self.sample_rate)
```

Comment:
Could you explain this change and the one above?

Reply:
The change above means that the raw strain (i.e. what directly comes out of the frame files) is only stored for the most recent 32 seconds instead of the entire length of what we store the *conditioned* strain for. We never actually need this buffer to go back to (or at least we don't at the moment as we only use the last chunks worth). The long buffer which was at 16834 took a *long* time to do the roll operation, which became a significant cost when you use shorter chunk sizes. 

The change highlighted here is to make sure that the *conditioned* strain is still the correct length as we do need that to be the full buffer worth, however it is at a much lower sample rate now and so isn't such a cost anyway to manipulate.